t_sparse_list.py

```diff
@@ -109,6 +109,17 @@ def test_slice_reversal_empty(self):
         sl = sparse_list.SparseList(4)
         self.assertEquals([None, None, None, None], sl[::-1])
 
+    def test_slice_with_list_read(self):
+        sl = sparse_list.SparseList([1, 2, 3, 4, 5])
+        ip = [0, 2, 4]
+        self.assertEquals([1, 3, 5], sl[ip])
```

Comment:
What's the use-case for this? My initial implementation mimicked the `list`'s interface. 

```python
>>> l = [1, 2, 3, 4, 5]
>>> ip = [0, 2, 4]
>>> l[ip]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: list indices must be integers, not list
```